api/src/test/java/org/pmiops/workbench/cohortreview/CohortReviewServiceImplTest.java

```diff
@@ -200,60 +200,97 @@ public void saveCohortReview() throws Exception {
-//        doNothing().when(mockPreparedStatement).setInt(5, 0);
-//        doNothing().when(mockPreparedStatement).setLong(6, participantCohortStatus.getParticipantKey().getCohortReviewId());
-//        doNothing().when(mockPreparedStatement).setLong(7, participantCohortStatus.getParticipantKey().getParticipantId());
-//        doNothing().when(mockPreparedStatement).addBatch();
-//        when(mockPreparedStatement.executeBatch()).thenReturn(new int[]{1});
-//        doNothing().when(mockPreparedStatement).close();
-//        doNothing().when(mockConnection).close();
-//
-//        cohortReviewService.saveParticipantCohortStatuses(pcsList);
-//
-//        verify(jdbcTemplate).getDataSource();
-//        verify(mockDatasource).getConnection();
-//        verify(mockConnection).setAutoCommit(false);
-//        verify(mockConnection).prepareStatement(compiledQuery);
-//        verify(mockPreparedStatement).setDate(1, participantCohortStatus.getBirthDate());
-//        verify(mockPreparedStatement).setLong(2, participantCohortStatus.getEthnicityConceptId());
-//        verify(mockPreparedStatement).setLong(3, participantCohortStatus.getGenderConceptId());
-//        verify(mockPreparedStatement).setLong(4, participantCohortStatus.getRaceConceptId());
-//        verify(mockPreparedStatement).setInt(5, 0);
-//        verify(mockPreparedStatement).setLong(6, participantCohortStatus.getParticipantKey().getCohortReviewId());
-//        verify(mockPreparedStatement).setLong(7, participantCohortStatus.getParticipantKey().getParticipantId());
-//        verify(mockPreparedStatement).addBatch();
-//        verify(mockPreparedStatement).executeBatch();
-//        verify(mockPreparedStatement).close();
-//        verify(mockConnection).close();
-//        verifyNoMoreMockInteractions();
+        DataSource mockDatasource = mock(DataSource.class);
+        Connection mockConnection = mock(Connection.class);
+        Statement mockStatement = mock(Statement.class);
+
+        List<ParticipantCohortStatus> pcsList = new ArrayList<>();
+        ParticipantCohortStatusKey key = new ParticipantCohortStatusKey(1, 1);
+        ParticipantCohortStatus participantCohortStatus = new ParticipantCohortStatus();
+        participantCohortStatus.setParticipantKey(key);
+        participantCohortStatus.setEthnicityConceptId(1L);
+        participantCohortStatus.setGenderConceptId(1L);
+        participantCohortStatus.setRaceConceptId(1L);
+        participantCohortStatus.setBirthDate(new Date(System.currentTimeMillis()));
+        pcsList.add(participantCohortStatus);
+
+        String sqlStatement = "insert into participant_cohort_status(" +
+                "birth_date, ethnicity_concept_id, gender_concept_id, race_concept_id, " +
+                "status, cohort_review_id, participant_id)" +
+                " values ('" + participantCohortStatus.getBirthDate().toString() + "', 1, 1, 1, 0, 1, 1)";
+
+        when(jdbcTemplate.getDataSource()).thenReturn(mockDatasource);
```

Comment:
Is there another way to test this, using JDBC with an inmemory DB or something?

Having all these mocks doesn't really tell us whether this works...